pkgs/desktops/pantheon/desktop/gala/default.nix

```diff
@@ -71,12 +73,34 @@ stdenv.mkDerivation rec {
     libcanberra-gtk3
     libgee
     mutter
+    systemd
   ];
 
   patches = [
     ./plugins-dir.patch
+    # Add systemd units so gnome-session can launch Pantheon
+    # using the systemd units instead of the XDG autostarts.
+    # https://github.com/elementary/gala/pull/1256
+    (fetchpatch {
+      url = "https://github.com/elementary/gala/commit/d425959aa2b9b9ac7ef622f601f3de1ca8ee768a.patch";
+      sha256 = "1r5jwmgvg970kz2yx79mz7cn6bfivsbqmi6s497q8z8lj8xrxr0g";
+    })
+    # autostarts: set hidden under systemd
+    # Prevents gnome-session from autostarting gala if systemd
+    # is launching the session instead.
+    # https://github.com/elementary/gala/pull/1255
+    (fetchpatch {
+      url = "https://github.com/elementary/gala/commit/bd362ec0c89bd72c5afca3f8313ee70d7ee25357.patch";
+      sha256 = "1l8dzyk7yncppg8yzhk7ddpccg3c0j95sar19hckc4wy4c1bn1j0";
+    })
+    # Hope this can make WITH_SYSTEMD block functions
+    # and make libsystemd found by pkg-config
+    # https://github.com/NixOS/nixpkgs/issues/101817#issuecomment-932956940
+    ./fix-gala-x11-timeout.patch
```

Comment:
The patch is accepted on upstream, however seems that I cannot change this part to `fetchpatch` as gala fail to build.

```nix
(fetchpatch {
  url = "https://github.com/elementary/gala/commit/cc8692ab2766d3fde42ebf8eaadeab3f5b3c45f7.patch";
  sha256 = "0c2nwawfdikqfcas4mffcp7klfl9wxxr3m5b7pxwl6iyycg4mwg2";
})
```

Related issue:

- https://github.com/NixOS/nixpkgs/issues/32084